kopykat-gradle-plugin/src/main/kotlin/fp/serrano/kopykat/KopyKatPlugin.kt

```diff
@@ -0,0 +1,97 @@
+package fp.serrano.kopykat
+
+import com.google.devtools.ksp.gradle.KspExtension
+import com.google.devtools.ksp.gradle.KspGradleSubplugin
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.kotlin.dsl.configure
+import org.gradle.kotlin.dsl.create
+import org.gradle.kotlin.dsl.withType
+import org.gradle.plugins.ide.idea.IdeaPlugin
+import org.gradle.plugins.ide.idea.model.IdeaModel
+import org.jetbrains.kotlin.gradle.dsl.kotlinExtension
+import org.jetbrains.kotlin.gradle.plugin.KotlinBasePlugin
+
+abstract class KopyKatPlugin : Plugin<Project> {
+
+    override fun apply(target: Project) {
+
+        val kkSettings = createKopyKatSettings(target)
+
+        configureKsp(target, kkSettings)
+
+        target.afterEvaluate {
+            // normally afterEvaluate should be avoided, but it's necessary here to allow KopyKatSettings to be
+            // configured otherwise applyKspPlugin will always be the default value.
+            if (kkSettings.applyKspPlugin.getOrElse(false)) {
+                target.pluginManager.apply("com.google.devtools.ksp")
+            }
```

Comment:
this doesn't seem to apply KSP automatically - I'm not sure why. I'll investigate.